lib/client/api.go

```diff
@@ -3020,8 +3024,23 @@ func makeProxySSHClient(ctx context.Context, tc *TeleportClient, sshConfig *ssh.
 	return client, nil
 }
 
+// CreatePROXYHeaderGetter returns PROXY headers signer with embedded client source/destination IP addresses,
+// which are taken from the context.
+func CreatePROXYHeaderGetter(ctx context.Context, proxySigner multiplexer.PROXYHeaderSigner) client.PROXYHeaderGetter {
+	if proxySigner != nil {
+		src, dst := utils.ClientAddrFromContext(ctx)
+		if src != nil && dst != nil {
+			return func() ([]byte, error) {
+				return proxySigner.SignPROXYHeader(src, dst)
+			}
+		}
+	}
```

Comment:
We could invert some of the conditions to reduce nesting. Like
```go
if proxySigner == nil {
    return nil
}
```

Reply:
Done.